src/api/store/userprofile.py

```diff
@@ -563,18 +563,77 @@ def update_user(self, context: Context, args) -> Tuple[dict, MassEnergizeAPIErro
   
   def delete_user(self, context: Context, user_id) -> Tuple[dict, MassEnergizeAPIError]:
     try:
-      if not user_id:
-        return None, InvalidResourceError()
-      
-      # check to make sure the one deleting is an admin
-      if not context.user_is_admin():
-        
-        # if they are not an admin make sure they can only delete themselves
-        if context.user_id != user_id:
-          return None, NotAuthorizedError()
+      if not self._has_access(context, user_id):
+        return None, CustomMassenergizeError("permission_denied")
       
       users = UserProfile.objects.filter(id=user_id)
-      users.update(is_deleted=True)
+      user = users.first()
+      old_email = user.email
+      new_email = "DELETED-" + datetime.today().strftime('%Y%m%d-%H%M') + "-" + old_email 
```

Comment:
why are we changing the email?

Reply:
Since we do not delete the record from the database, we modify the email address in case the person wants to create a profile again with that email.  This way we can tell exactly what happens in case needed.

Reply:
we should leave a comment on that line describing exactly what you just said here

Reply:
Done. Thanks for the request